### PR TITLE
chore(ci): fix our npm caching strategy for our e2e nextjs tests

### PR DESCRIPTION
**Pull request by JGAntunes**
**- Summary**

Similar to - https://github.com/netlify/cli/pull/3277 - for our e2e tests seems like we clone a total of 3 projects. We need to make use of the `cache-dependency-path`property to point to each of the lockfiles individually (see [here](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-dependencies)).

**- Test plan**

- https://github.com/netlify/zip-it-and-ship-it/runs/3588927992?check_suite_focus=true

**- A picture of a cute animal (not mandatory but encouraged)**

wombat-chore

### CODE REVIEW

File: .github/workflows/e2e-nextjs.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: Next.js Integration test
 
 on:
   pull_request:
-    branches: [main]
-    # Run on release PRs only
-    paths: ['CHANGELOG.md']
+    branches: [main, 'chore/fix-ci-npm-cache']
```
Comment (eduardoboucas) [inline .github/workflows/e2e-nextjs.yml:5]: Should we remove this change before merging?

File: .github/workflows/e2e-nextjs.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ name: Next.js Integration test
...
+    # Run on release PRs only, or changes to this same file
+    paths: ['CHANGELOG.md', '.github/workflows/e2e-nextjs.yml']
```
Comment (eduardoboucas) [inline .github/workflows/e2e-nextjs.yml:7]: Should we remove this change before merging?

Comment (eduardoboucas) [reply .github/workflows/e2e-nextjs.yml:7]: Actually, it kind of makes sense to run the CI when the workflow file changes, so maybe we should leave it?

Comment (JGAntunes) [reply .github/workflows/e2e-nextjs.yml:7]: If folks agree with it I would keep both these changes ☝️ similart to what we have with `fossa` -https://github.com/netlify/zip-it-and-ship-it/blob/main/.github/workflows/fossa.yml#L6-L7 - this way we have the ability to always test this during dev if needed be?

Comment (JGAntunes) [reply .github/workflows/e2e-nextjs.yml:7]: We could probably change the branch name though, maybe to something like `chore/ci-e2e-nextjs`? What do you think @eduardoboucas?

File: .github/workflows/e2e-nextjs.yml (hunk 1 above)
Comment (JGAntunes) [reply .github/workflows/e2e-nextjs.yml:5]: We could maybe keep this but under a different name, what do you think? 👇 

```suggestion
    branches: [main, 'chore/ci-e2e-nextjs']
```

File: .github/workflows/e2e-nextjs.yml (hunk 2 above)
Comment (eduardoboucas) [reply .github/workflows/e2e-nextjs.yml:7]: Oh, I see. So you'll use that branch whenever you want to test the flow? Sounds good to me!

**Review by eduardoboucas — APPROVED** [review]

**Pull request merged**